### PR TITLE
Resolved issue of wrong throttle values at spectrum chart when not all log parameters are selected in blackbox settings

### PR DESCRIPTION
**Pull request by demvlad**
The issues description:
The wrong throttle values at spectrum chart.

This issue appears if to disable some blackbox fields in list at Blackbox page of Betaflight Configurator 4.5 version.
For example - if enabled the gyro, gyro unfiltered and SetPoints only.
There is log file to test this.
[test.TXT](https://github.com/betaflight/blackbox-log-viewer/files/14496656/test.TXT)

The reason of this issue - the difference of logical conditions what are permissed creating of additional computing fields and filling data into it. 
(the functions buildFieldNames() and injectComputedFields() in flightlog.js file).
There are the incongruity fields indexes in log algoritmes in this case. As result - We are seeing the scaled yaw setpoint values instead of throttle in my example.
This PR resolved this issue.

### CODE REVIEW

File: js/flightlog.js [hunk 1]
```diff
@@ -657,29 +657,32 @@ function FlightLog(logData) {
...
 
-                    // Since version 4.0 is not more a virtual field. Copy the real field to the virtual one to maintain the name, workspaces, etc.
-                    if (sysConfig.firmwareType == FIRMWARE_TYPE_BETAFLIGHT  && semver.gte(sysConfig.firmwareVersion, '4.0.0')) {
-                        // Roll, pitch and yaw
-                        for (var axis = 0; axis <= AXIS.YAW; axis++) {
-                            destFrame[fieldIndex++] = srcFrame[setpoint[axis]];
-                        } 
-                        // Throttle
-                        destFrame[fieldIndex++] = srcFrame[setpoint[AXIS.YAW + 1]]/10;
+                    if(!that.isFieldDisabled().SETPOINT) {
```
Comment (haslinghuis) [inline js/flightlog.js:660]:
```suggestion
                    if (!that.isFieldDisabled().SETPOINT) {
```

File: js/flightlog.js [hunk 2]
```diff
@@ -657,29 +657,32 @@ function FlightLog(logData) {
...
 
-                    // Versions earlier to 4.0 we must calculate the expected setpoint
-                    } else {
-                        // Roll, pitch and yaw
-                        for (var axis = 0; axis <= AXIS.YAW; axis++) {
+                        // Since version 4.0 is not more a virtual field. Copy the real field to the virtual one to maintain the name, workspaces, etc.
+                        if (sysConfig.firmwareType == FIRMWARE_TYPE_BETAFLIGHT  && semver.gte(sysConfig.firmwareVersion, '4.0.0')) {
+                            // Roll, pitch and yaw
+                            for (var axis = 0; axis <= AXIS.YAW; axis++) {
```
Comment (haslinghuis) [inline js/flightlog.js:665]:
```suggestion
                            for (let axis = 0; axis <= AXIS.YAW; axis++) {
```

File: js/flightlog.js [hunk 3]
```diff
@@ -657,29 +657,32 @@ function FlightLog(logData) {
...
+                                destFrame[fieldIndex++] = srcFrame[setpoint[axis]];
+                            }
+                            // Throttle
+                            destFrame[fieldIndex++] = srcFrame[setpoint[AXIS.YAW + 1]]/10;
+
+                        // Versions earlier to 4.0 we must calculate the expected setpoint
+                        } else {
+                            // Roll, pitch and yaw
+                            for (var axis = 0; axis <= AXIS.YAW; axis++) {
```
Comment (haslinghuis) [inline js/flightlog.js:674]:
```suggestion
                            for (let axis = 0; axis <= AXIS.YAW; axis++) {
```

File: js/flightlog.js [hunk 4]
```diff
@@ -657,29 +657,32 @@ function FlightLog(logData) {
...
+                                destFrame[fieldIndex++] =
+                                    (rcCommand[axis] !== undefined ? that.rcCommandRawToDegreesPerSecond(srcFrame[rcCommand[axis]], axis, currentFlightMode) : 0);
```
Comment (haslinghuis) [inline js/flightlog.js:676]:
```suggestion
                                destFrame[fieldIndex++] = rcCommand[axis] !== undefined ? that.rcCommandRawToDegreesPerSecond(srcFrame[rcCommand[axis]], axis, currentFlightMode) : 0;
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Some minor changes. Otherwise LGTM.

**Review by haslinghuis — DISMISSED** [review]

**Comment by demvlad** [comment]
> Some minor changes. Otherwise LGTM.

Ok. Thank's.

**Comment by nerdCopter** [comment]
do you prefer `!( || )` or `( && )` for readability here? (two instances)

**Comment by haslinghuis** [comment]
Morgan's law :)

**Comment by demvlad** [comment]
> do you prefer `!( || )` or `( && )` for readability here? (two instances)

I prefer !a &&. !b
One moment, will change..

**Review by nerdCopter — DISMISSED** [review]
- approving, but only tested few logs for not-crashing and expected traces.
- not tested with logs missing parameters except the provided `test.TXT`.

**Review by haslinghuis — DISMISSED** [review]
Easier to read now :)

**Comment by nerdCopter** [comment]
> > do you prefer `!( || )` or `( && )` for readability here? (two instances)
> 
> I prefer !a &&. !b One moment, will change..

so the proper gate is `not-disabled-gyro && not-disabled-pid` ?  i think such does make sense.
is this for line 684 as well, i see the commit did not include such.

**Comment by demvlad** [comment]
> Easier to read now :)

morgan law is too clever:)

**Review by haslinghuis — APPROVED** [review]

**Comment by demvlad** [comment]
> * approving, but only tested few logs for not-crashing and expected traces.
> * not tested with logs missing parameters except the provided `test.TXT`.

Yes, this is a thin moment.
The index definition in one place and index using in other with some conditions. I've tryed to do identity conditions. But it is need to test this with different blackbox parameters set combination.

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**